### PR TITLE
Add Rectangular Satellite imagery

### LINKED ISSUE #601: Allow for rectangular regions of interest for satellite data
## Detailed Description
At the moment, we select square ROIs of satellite data (square in pixel space).

But, when projected to OSGB, those "square" images are actually quite tall rectangles. Which isn't ideal because, if anything, we probably want satellite images that are wider than taller (because most winds across the UK are from west to east)

A simple solution would be to allow for rectangular (in pixel space) satellite images, so we can manually request satellite imagery that is, say, 64 pixels tall and 128 pixels wide (which would be rectangular in pixel space, but might be very roughly square in OSGB).

Here are two plots of our satellite imagery at the moment:

Here's a plot in geostationary projection (which shows that the satellite image is square in pixel space):

image

And here's the exact same data, but now reprojected to OSGB. Note how tall and thin the satellite image is:

image

### PR DESCRIPTION
**Pull request by jacobbieker**
# Pull Request

## Description

Expands the configuration options for satellit imagery so that we can have non-square imagery, as the pixels are not square in shape. 

Fixes #601 

## How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration_

- [x] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_dataset/config/model.py [hunk 1]
```diff
@@ -313,7 +320,23 @@ class OpticalFlow(DataSourceMixin, TimeResolutionMixin):
...
+        ),
+    )
+    opticalflow_output_image_size_pixels_height: int = Field(
+        IMAGE_SIZE_PIXELS,
+        description=(
+            "The size of the images after optical flow has been applied. The output image is a"
+            " center-crop of the input image, after it has been flowed."
+        ),
+    )
+    opticalflow_input_image_size_pixels_width: int = Field(
```
Comment (peterdudfield) [inline nowcasting_dataset/config/model.py:339]: maybe update description here too?

File: nowcasting_dataset/config/model.py [hunk 2]
```diff
@@ -364,7 +388,8 @@ class GSP(DataSourceMixin, StartEndDatetimeMixin):
         description="The number of GSP samples per example. "
         "If there are less in the ROI then the data is padded with zeros. ",
     )
-    gsp_image_size_pixels: int = IMAGE_SIZE_PIXELS_FIELD
+    gsp_image_size_pixels_height: int = IMAGE_SIZE_PIXELS_FIELD
```
Comment (peterdudfield) [inline nowcasting_dataset/config/model.py:391]: suprised GSP uses image size?

File: nowcasting_dataset/data_sources/data_source.py [hunk 3]
```diff
@@ -372,14 +372,19 @@ class ImageDataSource(DataSource):
...
 
-    def __post_init__(self, image_size_pixels: int, meters_per_pixel: int):
+    def __post_init__(
+        self, image_size_pixels_height: int, image_size_pixels_width: int, meters_per_pixel: int
+    ):
         """Post Init"""
         super().__post_init__()
-        self._square = square.Square(
-            size_pixels=image_size_pixels, meters_per_pixel=meters_per_pixel
+        self._square = square.Rectangle(
```
Comment (peterdudfield) [inline nowcasting_dataset/data_sources/data_source.py:384]: _rectangle too?

File: nowcasting_dataset/square.py [hunk 4]
```diff
@@ -50,6 +50,44 @@ def bounding_box_centered_on(self, x_center_osgb: Number, y_center_osgb: Number)
...
+        Returns: Bounding box
+
+        """
+        return BoundingBox(
+            top=y_center_osgb + self._half_size_meters_height,
+            bottom=y_center_osgb - self._half_size_meters_height,
+            left=x_center_osgb - self._half_size_meters_width,
+            right=x_center_osgb + self._half_size_meters_width,
+        )
+
```
Comment (peterdudfield) [inline nowcasting_dataset/square.py:89]: nice

File: tests/config/test_no_hrv.yaml [hunk 5]
```diff
@@ -21,7 +22,8 @@ input_data:
   satellite:
     satellite_channels:
       - IR_016
-    satellite_image_size_pixels: 24
+    satellite_image_size_pixels_height: 24
```
Comment (peterdudfield) [inline tests/config/test_no_hrv.yaml:25]: worth putting this as different values just tot check

File: tests/config/test_old_pv.yaml [hunk 6]
```diff
@@ -19,12 +20,14 @@ input_data:
...
       - IR_016
-    satellite_image_size_pixels: 24
+    satellite_image_size_pixels_height: 24
+    satellite_image_size_pixels_width: 24
     satellite_zarr_path: tests/data/sat_data.zarr
   hrvsatellite:
     hrvsatellite_channels:
       - HRV
-    hrvsatellite_image_size_pixels: 64
+    hrvsatellite_image_size_pixels_height: 64
```
Comment (peterdudfield) [inline tests/config/test_old_pv.yaml:29]: different values?

File: tests/manager/conftest.py [hunk 7]
```diff
@@ -38,7 +39,8 @@ def hrvsat():
         zarr_path=filename,
         history_minutes=30,
         forecast_minutes=60,
-        image_size_pixels=64,
+        image_size_pixels_height=64,
```
Comment (peterdudfield) [inline tests/manager/conftest.py:42]: same, maybe try some different values just to be sure

File: nowcasting_dataset/config/model.py (hunk 2 above)
Comment (jacobbieker) [reply nowcasting_dataset/config/model.py:391]: Yeah, quite a few inherit from ImageDataSource, which is where I ended up changing the code.

**Pull request merged**